tree/avl-tree/avl-tree.js

```diff
@@ -0,0 +1,66 @@
+const { BinarySearchTree } = require('../binary-search-tree/binary-search-tree')
+const { BinarySearchTreeNode } = require('../binary-search-tree/binary-search-tree-node')
+
+class AVLTree extends BinarySearchTree {
+  // Insert the given data into tree
+  insert(value) {
+    super.insert(value)
+    let curNode = this.root.find(value)
+
+    while (curNode) {
+      this.balance(curNode)
+      curNode = curNode.parent
+    }
```

Comment:
The balance check can be made while the recursive calls exit. By creating a loop, you recursively go down the tree, then go up, and then go another loop again.